### DISCUSSION TITLE
[Proposal] Enums and matching

### DISCUSSION THREAD

**Discussion by alshdavid** [alshdavid/TypeScriptBC#79] [Ideas]
TypeScript has enums with assignable values. In this context how would we support Rust-like enum matching capabilities

```typescript
enum Foo {
  A = 'A',
  B = 'B',
}
```

**Comment by agentcooper** [comment]
On a topic related to enums and matching, I think there is a possibility to improve on how algebraic data types work in BorrowScript compared to TypeScript, please see https://github.com/microsoft/TypeScript/issues/36336.

**Comment by SuperSonicHub1** [comment]
The suggested syntax looks very much not like JS, and what's the difference
between this and mix-ins/inheritance?
